apiary-metastore-events/apiary-hive-events/src/main/java/com/expediagroup/apiary/extensions/events/metastore/event/ApiaryInsertEvent.java

```diff
@@ -34,13 +37,26 @@
 
   public ApiaryInsertEvent(InsertEvent event) {
     super(event);
-    databaseName = event.getDb();
-    tableName = event.getTable();
-    partitionKeyValues = event.getPartitionKeyValues();
```

Comment:
event.getPartionKeyValues is no longer a method in hive 3. We have to include our own similar logic to keep this functionality.

Reply:
Aah no, they've even made these `HiveEvent` objects not backwards compatible. That makes the whole listener thing really tricky (impossible?) to do across Hive 2 and 3. :(

Reply:
yeah it seems like we'd either need to switch or take a really careful approach to shim in functionality / compatibility.